rules.py

```diff
@@ -274,27 +346,29 @@ def check(self, element):
 class GpUnitOcdId(ElectoralDistrictOcdId):
     """Any GpUnit that is a geographic district SHOULD have a valid OCD-ID."""
 
-    districts = ["borough", "city", "county", "municipality","state", "town",
-                "township", "village"]
+    districts = [
+        "borough", "city", "county", "municipality", "state", "town",
+        "township", "village"
+    ]
+    validate_ocd_file = True
 
     def __init__(self, election_tree, schema_file):
         super(GpUnitOcdId, self).__init__(election_tree, schema_file)
-        self.ocds = self._get_ocd_data()
 
     def elements(self):
         return ["ReportingUnit"]
 
     def check(self, element):
         gpunit_type = element.find("Type")
-        if gpunit_type.text in self.districts:
+        if gpunit_type is not None and gpunit_type.text in self.districts:
             for extern_id in element.iter("ExternalIdentifier"):
                 id_type = extern_id.find("Type")
                 if id_type is not None and id_type.text == "ocd-id":
                     value = extern_id.find("Value")
                     if value.text not in self.ocds:
                         raise base.ElectionWarning(
-                        "The OCD ID %s defined on line %d is not "
-                        "valid" % (value.text, value.sourceline))
+                            "The OCD ID %s defined on line %d is not "
+                            "valid" % (value.text, value.sourceline))
```

Comment:
Please update this error message to be what's currently in the master branch. Thanks.